### ISSUE TITLE
Why tensorflow don't support tf.float64 on some ops?

### ISSUE THREAD

**Issue by gauss-clb** [tensorflow/tensorflow#15038]
Reproducing experiment of some paper using tensorflow may don't have good performance. I have tried my best to keep the structure and hyper-parameters unchanged. Does anyone encounter similar problem? Is it the code wrong?  I wonder that `caffe` can use `float64` but tensorflow only can use `float32`,  so there are more inaccuracy on `gradients`, why tf don't support `tf.float64`?

**Comment by aselle** [comment]
Most model developers try to use the minimum precision necessary. Many (most?) models can be done with float32, and it is generally much faster. Adding float64 support everywhere doubles the amount of code to maintain and to have in a binary. That being said, if there is a strong reason, we would be happy to support float64 in spaces as needed.

**Comment by gauss-clb** [comment]
I just want to reproducing experiment of some paper, but I can't get good performance. I think it maybe the error of gradient due to `tf.float32`, such as `convolution` ops.  I wonder that small numerical error will accumulate and get bigger error, like butterfly effect.

Like [DeepTextures](https://github.com/leongatys/DeepTextures), the author use `vgg19` of `caffe` to calculate gradient of loss and use `[L-BFGS-B](https://github.com/leongatys/DeepTextures/blob/master/DeepImageSynthesis/ImageSyn.py#L54)` of  `scipy.optimize` and get [good performance](https://github.com/leongatys/DeepTextures/blob/master/Example.ipynb). 

But I try to use `vgg19` of `tensorflow` to calculate gardient and `L-BFGS-B`,  it convergence so fast and the performance is so bad. Maybe I think the precision of gradient is not enough and it cause `L-BFGS-B` convergence so fast.

**Comment by gauss-clb** [comment]
@aselle, it is `tf.control_dependencies` causes `L-BFGS-B` converge so fast. It is hard to find this problem, see my [new issue](https://github.com/tensorflow/tensorflow/issues/15077). I'm very puzzled, the behavior of `tf.control_dependencies`  is so weird.

**Comment by drpngx** [comment]
@gauss-clb I am going to close this. Your initial ask about float64 support has been heard. For questions about numerical issues with neural networks, I suggest asking on stackoverflow.
